File: LeetCode/JavaScript/0075-Sort-Colors.js
// 75. Sort Colors
// https://leetcode.com/problems/sort-colors/

/**
 * @param {number[]} nums
 * @return {void}
 */

var sortColors = function (nums) {
  const numsLength = nums.length;
  const countingArray = Array(3).fill(0);
  for (let i = 0; i < numsLength; i++) countingArray[nums[i]]++;

  let i = 0,
    nextIndex = 0;
  while (i < numsLength) {
    if (countingArray[nextIndex] !== 0) {
      nums[i] = nextIndex;
      countingArray[nextIndex]--;
      i++;
    } else nextIndex++;
  }
};

File: LeetCode/Python/0075_Sort_Colors.py
# 75. Sort Colors
# https://leetcode.com/problems/sort-colors/


class Solution:
    def sortColors(self, nums: list[int]) -> None:
        i = 0
        j = 0
        temp = 0
        size = len(nums)
        for i in range(0, size):
            for j in range(0, size - 1 - i):
                if nums[j] > nums[j + 1]:
                    temp = nums[j + 1]
                    nums[j + 1] = nums[j]
                    nums[j] = temp
